plugins/providers/alicloudiam/config.go

```diff
@@ -0,0 +1,228 @@
+package alicloudiam
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/bearaujus/bptr"
+	"github.com/go-playground/validator/v10"
+	"github.com/goto/guardian/domain"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	AccountTypeRamUser = "ramUser"
+	AccountTypeRamRole = "ramRole"
+)
+
+type Credentials struct {
+	AccessKeyID     string `mapstructure:"access_key_id" json:"access_key_id" validate:"required,base64"`
+	AccessKeySecret string `mapstructure:"access_key_secret" json:"access_key_secret" validate:"required,base64"`
+	RoleToAssume    string `mapstructure:"role_to_assume" json:"role_to_assume,omitempty"`
+	ResourceName    string `mapstructure:"resource_name" json:"resource_name" validate:"required"`
```

Comment:
what is this needed for?

Reply:
![image](https://github.com/user-attachments/assets/8927913b-ce0c-4dcf-8c63-be3e45187b78)
Actually for constructing this fields for the resource URN ^

But yup I just following gcloudiam template for this.. Do you have any other preference for constructing the resource URN @rahmatrhd ? 🙏

cc: @Gadigeppa-J

Reply:
we can replace the resource name with the account_id number

Reply:
Resolved: https://github.com/goto/guardian/pull/190/commits/6bd8fdffe0d3723f936bc84ddaf326d111b06d9b